index.js

```diff
@@ -1,39 +1,99 @@
 async function setupPlugin({config, global}) {
-    global.posthogUrl = config.postHogUrl
+    global.posthogUrl = cleanInstanceUrl(config.postHogUrl)
     global.apiToken = config.postHogApiToken
     global.projectToken = config.postHogProjectToken
-    global.syncScoresIntoPosthog = global.posthogUrl && global.apiToken && global.projectToken
+    global.zapierUrl = config.zapierUrl
+    global.actionIdToName = {}
+    config.actionIdToName.split(',').forEach(pair => {
+        const [actionId, conversionName] = pair.split(':')
+        global.actionIdToName[parseInt(actionId)] = conversionName
+    })
+    global.defaultStartTime = config.defaultStartTime ? new Date(config.defaultStartTime) : new Date(2021, 6, 1)
+    global.isConfigSet = global.posthogUrl
+        && global.apiToken
+        && global.projectToken
+        && global.zapierUrl
+        && Object.keys(global.actionIdToName).length
+        && isValidDate(global.defaultStartTime)
 }
 
+function cleanInstanceUrl(url) {
+    return url.replace(/\/$/, '')
+}
+
+function isValidDate(date) {
+    return date.toString() !== 'Invalid Date'
+}
 
 function addDays(date, days) {
-    var result = new Date(date);
-    result.setDate(result.getDate() + days);
-    return result;
+    const result = new Date(date)
+    result.setDate(result.getDate() + days)
+    return result
+}
+
+function formatTimestampForGoogle(date){
+    const result = new Date(date)
+    if (isValidDate(result)) {
+        result.setMilliseconds(0)
+        return result.toISOString().replace(/\.\d+Z$/, '+0000')
+    } else {
+        console.warn(`Received invalid date "${date}"`)
+        return date
+    }
 }
 
-async function runEveryMinute({config, global, storage}) {
-    if (!global.syncScoresIntoPosthog) {
-        console.log('Not syncing Hubspot Scores into PostHog - config not set.')
+function extractGclidFromEvent(event) {
+    const eventProperties = event.properties || {}
+    const personProperties = event.person?.properties || {}
+
+    if ('gclid' in eventProperties) {
+        return eventProperties.gclid
+    }
+    if ('gclid' in personProperties) {
+        return personProperties.gclid
+    }
+    if ('$initial_gclid' in personProperties) {
+        return personProperties.$initial_gclid
+    }
+    if ('$set' in eventProperties) {
+        const { $set } = eventProperties
+        if ('gclid' in $set) {
+            return $set.gclid
+        }
+        if ('$initial_gclid' in $set) {
+            return $set.$initial_gclid
+        }
+    }
+    if ('$set_once' in eventProperties) {
+        const { $set_once } = eventProperties
+        if ('gclid' in $set_once) {
+            return $set_once.gclid
+        }
+        if ('$initial_gclid' in $set_once) {
+            return $set_once.$initial_gclid
+        }
+    }
+
+    return null
+}
+
+async function runEveryMinute({ global, storage }) {
+    if (!global.isConfigSet) {
```

Comment:
I'm thinking we should just raise an error in `setupPlugin` instead of handling this here since there isn't any other use to this plugin?

Initially it was done this way for the Hubspot sync since that plugin supports multiple things (syncing into hubspot and syncing from hubspot) that didn't have the same config requirements across the function calls.